renderer.php

```diff
@@ -917,23 +917,36 @@ private function construct_fullname_head($data) {
         global $CFG;
 
         $url = $data->url();
-        if ($data->pageparams->sort == ATT_SORT_LASTNAME) {
+        if ($data->pageparams->sort == ATT_SORT_IDNUMBER) {
             $url->param('sort', ATT_SORT_FIRSTNAME);
             $firstname = html_writer::link($url, get_string('firstname'));
+            $url->param('sort', ATT_SORT_LASTNAME);
+            $lastname = html_writer::link($url, get_string('lastname'));
+            $idnumber = get_string('idnumber');
+        } else if ($data->pageparams->sort == ATT_SORT_LASTNAME) {
             $lastname = get_string('lastname');
+            $url->param('sort', ATT_SORT_IDNUMBER);
+            $idnumber = html_writer::link($url, get_string('idnumber'));
+            $url->param('sort', ATT_SORT_FIRSTNAME);
+            $firstname = html_writer::link($url, get_string('firstname'));
         } else if ($data->pageparams->sort == ATT_SORT_FIRSTNAME) {
             $firstname = get_string('firstname');
             $url->param('sort', ATT_SORT_LASTNAME);
             $lastname = html_writer::link($url, get_string('lastname'));
+            $url->param('sort', ATT_SORT_IDNUMBER);
+            $idnumber = html_writer::link($url, get_string('idnumber'));
         } else {
+	        $idnumber = html_writer::link($data->url(array('sort' => ATT_SORT_IDNUMBER)), get_string('idnumber'));
             $firstname = html_writer::link($data->url(array('sort' => ATT_SORT_FIRSTNAME)), get_string('firstname'));
             $lastname = html_writer::link($data->url(array('sort' => ATT_SORT_LASTNAME)), get_string('lastname'));
         }
 
-        if ($CFG->fullnamedisplay == 'lastname firstname') {
-            $fullnamehead = "$lastname / $firstname";
+        if ($CFG->fullnamedisplay == 'idnumber firstname lastname') {
```

Comment:
this all feels a bit to specific to your use of idnumber - usually people would drop it in a new column rather than in the user fullname field - I think we should avoid these changes and just add the sort option.

but... I'm also hoping post 4.0 release we can re-do a chunk of the user reports using reportbuilder code which will hopefully give us some better control over columns/sorting etc.

Reply:
Yes, you are right and adding the sort option would be great.